modules/executors/main.tf

```diff
@@ -1,14 +1,21 @@
 locals {
-  prefix = var.resource_prefix != "" ? "${var.resource_prefix}-sourcegraph-" : "sourcegraph-"
+  network_tags = [
+    substr("${random_id.compute_instance_network_tag.hex}-executors", 0, 64),
+    var.instance_tag,
+    "executors"
+  ]
 }
 
 # Fetch the google project set in the currently used provider.
 data "google_project" "project" {}
 
+resource "random_id" "service_account" {
+  prefix      = var.resource_prefix
+  byte_length = 4
+}
 resource "google_service_account" "sa" {
-  # ID can be no longer than 28 characters.
-  account_id   = "${substr(local.prefix, 0, 19)}executors"
-  display_name = "${var.resource_prefix}${var.resource_prefix != "" ? " " : ""}sourcegraph executors"
+  account_id   = substr("${random_id.service_account.hex}-executors", 0, 30)
```

Comment:
can this ever reach 30 characters if the rand id is only 4 bytes? Asking because the previous comment suggests that only 28 characters total would be allowed.

Reply:
The constraint as mentioned in the [docs](https://cloud.google.com/resource-manager/docs/creating-managing-projects) is 30 characters.